sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuFileFormatDataWriter.scala

```diff
@@ -409,6 +505,340 @@ class GpuDynamicPartitionDataWriter(
   }
 }
 
+/**
+ * Dynamic partition writer with concurrent writers, meaning multiple concurrent writers are opened
+ * for writing.
+ *
+ * The process has the following steps:
+ *  - Step 1: Maintain a map of output writers per each partition columns. Keep all
+ *    writers opened; Cache the inputted batches by splitting them into sub-groups and
+ *    each partition holds a list of sub-groups; Find and write the the max pending
```

Comment:
nit: `the the` should probably just be `the`

Reply:
Done